src/topology/separation.lean

```diff
@@ -150,6 +150,33 @@ instance subtype.t1_space {α : Type u} [topological_space α] [t1_space α] {p
 instance t1_space.t0_space [t1_space α] : t0_space α :=
 ⟨λ x y h, ⟨{z | z ≠ y}, is_open_ne, or.inl ⟨h, not_not_intro rfl⟩⟩⟩
 
+lemma t1_characterisation : t1_space α ↔
+  ∀ (x y), x ≠ y → (∃ (U : set α) (hU : is_open U), x ∈ U ∧ y ∉ U) :=
+begin
+  split,
+  { introI t1,
+    intros x y hxy,
+    exact ⟨{y}ᶜ, is_open_compl_iff.mpr (t1_space.t1 y),
+            mem_compl_singleton_iff.mpr hxy,
+            not_not.mpr rfl⟩},
+  { intro h,
+    fconstructor,
+    intro x,
+    fconstructor,
+    have p : ⋃₀ {U : set α | (x ∉ U) ∧ (is_open U)} = {x}ᶜ,
+    { apply subset.antisymm; intros t ht,
+      { rcases ht with ⟨A, ⟨hxA, hA⟩, htA⟩,
+        rw [mem_compl_eq, mem_singleton_iff],
+        intro htx,
+        subst htx,
+        contradiction },
+      { obtain ⟨U, hU, hh⟩ := h t x (mem_compl_singleton_iff.mp ht),
+        exact ⟨U, ⟨hh.2, hU⟩, hh.1⟩}},
+    rw ← p,
+    have c : ∀ B ∈ {U : set α | (x ∉ U) ∧ (is_open U)}, is_open B, by finish,
+    exact is_open_sUnion c}
```

Comment:
```suggestion
    rw ← p,
    exact is_open_sUnion (λ B hB, hB.2)}
```
@ericrbg this is the change I was talking about. I think now the suggestion it's well-done.

Reply:
yep that's probably best! no worries about it being your first PR; mine was a month ago and you get used to this chaos ;b

Reply:
(also, if you go to [the files changed](https://github.com/leanprover-community/mathlib/pull/7367/files) menu, you can commit a bunch of suggestions at once instead of having one commit/suggestion)

Reply:
Thanks, I commit the changes!

Reply:
Thank you very much for the suggestions and advices. Next time I'll try to commit suggestions from the files changed menu.